dart/lib/src/protocol/sentry_exception.dart

```diff
@@ -0,0 +1,59 @@
+import 'package:meta/meta.dart';
+
+import 'mechanism.dart';
+import 'sentry_stack_trace.dart';
+
+/// The Exception Interface specifies an exception or error that occurred in a program.
+class SentryException {
```

Comment:
why do we have final fields in all the data classes but not this one?

Reply:
In java this class seems mutable and hasn't a `clone()` method, so I translated like this. Should we make it immutable? and add a `clone()` ? or `copyWith()` ?

Reply:
SentryEvent on Java is mutable and it is immutable on Dart, we'd need to go the way of what makes more sense to the language

Reply:
✅  `SentryException` and `SentryStackFrame` are now immutable.